### PR TITLE
Regenerate the gitwash docs.

### LINKED ISSUE #8123: Regenerate gitwash docs for improved syntax highlighting of non-python code sections
See https://github.com/matthew-brett/gitwash/issues/10.

### PR DESCRIPTION
**Pull request by aashil**
Regenerate the gitwash docs and add README.txt documenting how to use the script to generate the docs. Fixes #8123

### CODE REVIEW

File: doc/devel/README.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+All documentation in the gitwash directory are automatically generated by running the below script in
+the project's root directory:
+
+python gitwash_dumper.py doc/devel Matplotlib --repo-name=matplotlib --github-user=matplotlib \
```
Comment (NelleV) [inline doc/devel/README.txt]: Can you add a link to the gitwash script?

**Comment by NelleV** [comment]
Apart from my comment, it looks good to me.

File: doc/devel/README.txt (hunk 1 above)
Comment (aashil) [inline doc/devel/README.txt]: I restructured the sentence a bit and added the link to gitwash_dumper.py script. Have a look.

**Comment by QuLogic** [comment]
Documentation isn't building:
> Warning, treated as error:
> 
> /home/travis/build/matplotlib/matplotlib/doc/devel/gitwash/setting_up_for_development.rst:: WARNING: document isn't included in any toctree

**Comment by matthew-brett** [comment]
It looks like the `setting_up_for_development` page is a version of gitwash's `git_development` page, and can be deleted.

**Comment by QuLogic** [comment]
Still more doc errors:
> Warning, treated as error:
>
> /home/travis/build/matplotlib/matplotlib/doc/devel/gitwash/dot2_dot3.rst:: WARNING: document isn't included in any toctree

**Comment by aashil** [comment]
Any idea why appveyor and codecov fail?

**Comment by phobson** [comment]
@aashil they're both kink of finiky. I looks like appveyor had some transient network issues. I restarted its build.

**Comment by dstansby** [comment]
The fancy log bit is over-running the text box quite a bit, does anyone know if there's a way to fix this?
screen shot 2017-02-25 at 12 44 54

**Comment by QuLogic** [comment]
It's a literal block, so I think the only choice is to pick a shorter set of log lines.

**Comment by dstansby** [comment]
Not sure if this wants a backport - if anyone else thinks it does feel free to do the backport.

**Comment by tacaswell** [comment]
Backported to v2.0.0-doc as 1d87fbae6

**Pull request merged**